packages/console/src/pages/Connectors/components/ConnectorForm/hooks.tsx

```diff
@@ -0,0 +1,26 @@
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+
+import { safeParseJson } from '@/utilities/json';
+
+export const useConfigParser = () => {
```

Comment:
Make this a more general useSafeParseJson hook?  Find a  similar logic used in the `UserDetails/UserSettings `page as well.

Reply:
This is a wrapper for "safeParseJson", add some error messages and additional checks.